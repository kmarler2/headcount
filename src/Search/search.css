input {
  width: 40%;
  height: 30px;
  margin-left: 26.5%;
  font-size: 25px;
  padding: 5px;
  margin-bottom: 60px;
  box-shadow: 3px 3px 0px 0px gray;
}

input::placeholder {
  font-size: 25px;
}

input:focus {
  outline: none;
}

.button {
  height: 45px;
  padding: 5px;
  font-size: 25px;
  color: #F7F7FF;;
  background-color: #BDD4EA;
  box-shadow: 3px 3px 0px 0px gray;
}