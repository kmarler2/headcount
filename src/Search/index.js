import React, { Component } from 'react';
import './search.css';
import PropTypes from 'prop-types';

class Search extends Component {
  
  handleChange = (event) => {
    this.props.findDistrict(event.target.value);
  }

  render() {
    return (
      <form>
        <input
          className="input-form"
          placeholder='Search for a district'
          type="text"
          onChange={ this.handleChange } />
      </form>
    );
  }
}

Search.propTypes = {
  findDistrict: PropTypes.func
};

export default Search;