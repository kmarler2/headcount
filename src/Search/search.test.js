import React from 'react';
import Search from './index';
import { shallow } from 'enzyme';

describe('search', () => { 
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(wrapper=(<Search />));
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});