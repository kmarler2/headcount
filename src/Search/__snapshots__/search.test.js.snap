// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="input-form"
        onChange={[Function]}
        placeholder="Search for a district"
        type="text"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "input-form",
        "onChange": [Function],
        "placeholder": "Search for a district",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="input-form"
          onChange={[Function]}
          placeholder="Search for a district"
          type="text"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-form",
          "onChange": [Function],
          "placeholder": "Search for a district",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
